src/components/Form/Form.tsx

```diff
@@ -68,6 +68,7 @@ const Form = (props: FormProps) => {
 			let values: MosaicObject;
 
 			if (getFormValues) {
+				await dispatch(formActions.disableForm(true));

```

Comment:
I think we might want to disable the form before this if statement, this way it can be disabled even while assigning values through the defaultValue prop, then it'll get enabled back again at the end.